x-pack/legacy/plugins/infra/public/pages/logs/analysis/sections/helpers/ml_links.tsx

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { encode } from 'rison-node';
+
+export type GetMlLink = (partitionId?: string | undefined) => string;
+
+export const getMlLinkFormatter = ({
+  basePath,
+  startTime,
+  endTime,
+  jobId,
+}: {
+  basePath: string;
+  startTime: number;
+  endTime: number;
+  jobId: string;
+}) => (partitionId?: string) => {
+  const startTimeParam = new Date(startTime).toISOString();
+  const endTimeParam = new Date(endTime).toISOString();
+
+  const _g = encode({
+    ml: {
+      jobIds: [jobId],
+    },
+    time: {
+      from: startTimeParam,
+      to: endTimeParam,
+      mode: 'absolute',
+    },
+  });
+
+  const baseLink = `${basePath}/app/ml#/timeseriesexplorer?_g=${_g}`;
```

Comment:
In other places we seem to use `url.format` to assemble the urls - maybe that would be a good idea here too?

https://github.com/elastic/kibana/blob/3d28467d000d1a612e6a3ef756f00cd27d51bdf9/x-pack/legacy/plugins/infra/public/components/logging/log_entry_flyout/log_entry_actions_menu.tsx#L124-L129